components/http/okHttp/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java

```diff
@@ -899,6 +900,31 @@ public MediaType contentType() {
                                     }
                                 }
 
+                                @Override
+                                public long contentLength() {
+                                    long length;
+                                    final Set<String> contentLength =
+                                            requestInfo.headers.getOrDefault(
+                                                    contentLengthHeaderKey, new HashSet<>());
+                                    if (contentLength.isEmpty()) {
+                                        try (final ByteArrayInputStream contentStream =
+                                                (ByteArrayInputStream) requestInfo.content) {
+                                            length = contentStream.available();
+                                        } catch (IOException ex) {
+                                            length = -1L;
+                                        }
```

Comment:
```suggestion
                                    if (contentLength.isEmpty() && requestInfo.content instanceof ByteArrayInputStream) {
                                        final ByteArrayInputStream contentStream = (ByteArrayInputStream) requestInfo.content;
                                        try  {
                                            length = contentStream.available();
                                        } catch (IOException ex) {
                                            length = -1L;
                                        }
```

we don't want to be disposing the stream here since it's already taken care of by okhttp down the pipeline
we also want to type test before we cast to avoid having things blow up if it's not the expected type.